### PR TITLE
Update APM Server documentation for use since 8.x requiring Kibana.

### PR DESCRIPTION
**Pull request by naemono**
related: #7541 

Since Kibana is now a requirement for APM Server since 8.x this updates the documentation appropriately.

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 1]
```diff
@@ -40,9 +40,25 @@ spec:
...
   elasticsearchRef:
     name: quickstart
+  kibanaRef: <1>
+    name: quickstart
 EOF
 ----
 
+<1> A reference to a Kibana instance is only required for APM Server versions 8.0.0 and later.
+
+NOTE: If Kibana is referenced in the APM Server specification then the following configuration is required in the Kibana specification to install the APM ILM templates:
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:50]: Is that true? It is only required if you are using version 8.0.0 and later. So maybe: something along the lines of: "starting with version 8.0.0 the following Kibana configuration is requireed to run APM Server"

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 2]
```diff
@@ -106,6 +122,18 @@ spec:
 EOF
 ----
 
+You will also need to ensure that your Kibana instance has the following configuration:
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:125]:
```suggestion
Starting with version 8.0.0 you will also need to ensure that your Kibana instance has the following configuration:
```

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 3]
```diff
@@ -23,11 +23,11 @@ This section describes how to deploy, configure and access an APM Server with EC
 [id="{p}-apm-eck-managed-es"]
 == Use an Elasticsearch cluster managed by ECK
 
-Managing both APM Server and Elasticsearch by ECK allows a smooth and secured integration between the two. The output configuration of the APM Server is setup automatically to establish a trust relationship with Elasticsearch.
+Managing the APM Server a Kibana instance and Elasticsearch with ECK allows a smooth and secured integration between the stack components. The output configuration of the APM Server is setup automatically to establish a trust relationship with Elasticsearch. Specifying the Kibana reference allows ECK to automatically configure the Kibana endpoint as described in https://www.elastic.co/guide/en/apm/server/current/setup-kibana-endpoint.html
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:26]: I find this first sentence a bit hard to parse. Not sure if I have a better suggestion though. 
```suggestion
Managing APM Server, Kibana and Elasticsearch with ECK allows a smooth and secured integration between the stack components. The output configuration of the APM Server is setup automatically to establish a trust relationship with Elasticsearch. Specifying the Kibana reference allows ECK to automatically configure the  link:https://www.elastic.co/guide/en/apm/server/current/setup-kibana-endpoint.html[Kibana endpoint].
```

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc (hunk 1 above)
Comment (pebrc) [reply docs/orchestrating-elastic-stack-applications/apm-server.asciidoc]: Yes but my comment was more about the wording you had. It implies that this is required whenever a `kibanaRef` is specified, when in fact we should qualify this with "starting with 8.0.0"

Comment (naemono) [reply docs/orchestrating-elastic-stack-applications/apm-server.asciidoc]: @pebrc My comment may have been misleading, as I had also updated the wording, in additional to double checking the requirement.

**Review by pebrc — APPROVED** [review]
